### PR TITLE
Unify Fido2 authentication under two-factor tab

### LINKED ISSUE #2754: Unify Fido2 authentication under two-factor tab
**Description**
As discussed on Mattermost UX/UI
In BTCPay Server user settings (`/Manage/Index`), currently, we have two tabs:

- Two Factor Authentication 
- FIDO2 Authentication

Screenshot 2021-08-07 at 10 40 04

We should probably unify them in a single Two Factor Authentication view and then offer two options in the same way the current `Two Factor Authentication page does.

Something like: (though precise text can be discussed)

- Two-step log in
   - Authenticator App (Authenticate with 6-digit PIN provided by apps such as Google or Microsoft Authenticator)
   - Security Key (Authenticate with your hardware wallet supporting FIDO2 or a USB key like - Yubikey) 

Screenshot 2021-08-07 at 10 58 07

Relevant Mattermost discussion here https://chat.btcpayserver.org/btcpayserver/pl/z4n517jrn7fhjk57cqtz6omeir

### PR DESCRIPTION
**Pull request by dennisreimann**
Closes #2754.

I'd say we should refine and simplify the wording -- getting rid of technical terms like FIDO2 will make it easier for the user to understand what's needed and what might be the options here. Input welcome.

01-start

02-setup

### CODE REVIEW

**Comment by Kukks** [comment]
Concept ACK. Much cleaner and easier to explain this way

**Comment by dennisreimann** [comment]
Updated the UI and wording a bit. If you want to, you can take a look and maybe improve the copy @Zaxounette.

**Comment by pavlenex** [comment]
Great improvements. 

Few notes
1. I'd just somehow make this page a bit more compact:

Screenshot 2021-09-09 at 18 36 34

2. If I could neat-pick, since these are two distinct methods of securing the account, I'd separate them on separate views/pages.

Screenshot 2021-09-09 at 18 44 24
- Enable 2FA
Using apps such as Google or Microsoft Authenticator.
- Enable Security device
A physical device, or a hardware wallet is used for signing in.
(Google explains it similarly)
Screenshot 2021-09-09 at 18 39 00


3. Just some google screens for inspiration on flow, wording.

Screenshot 2021-09-09 at 18 39 28

Screenshot 2021-09-09 at 18 41 29


Of all 3, 1 is only something I feel strongly we should improve, the other two are more suggestions, this is already a big improvement.

**Comment by pavlenex** [comment]
@dennisreimann Also, this is probably my caching, but I noticed I cannot switch themes in the header menu, so please try it out, I'm sure it'll work on your end though.

**Comment by dennisreimann** [comment]
I think this is related to the CSP changes. Will take a closer look tomorrow. Out of office now. 😉

File: BTCPayServer/Views/Manage/TwoFactorAuthentication.cshtml [hunk 1]
```diff
@@ -3,83 +3,123 @@
...
+            {
+                <a asp-action="Disable2fa" class="list-group-item d-flex justify-content-between align-items-center list-group-item-action py-3" data-bs-toggle="modal" data-bs-target="#ConfirmModal" data-title="Disable two-factor authentication (2FA)" data-description="Disabling 2FA does not change the keys used in the authenticator apps. If you wish to change the key used in an authenticator app you should reset your authenticator keys." data-confirm="Disable" data-confirm-input="DISABLE">
+                    <div>
+                        <h5>Disable 2FA</h5>
+                        <p class="mb-0 me-3">Re-enabling will not require you to reconfigure your app.</p>
+                    </div>
+                    <vc:icon symbol="caret-right" />
+                </a>
+                <a asp-action="GenerateRecoveryCodes" class="list-group-item d-flex justify-content-between align-items-center list-group-item-action py-3" data-bs-toggle="modal" data-bs-target="#ConfirmModal" data-title="Reset recovery codes" data-description="Your existing recovery codes will no longer be valid!" data-confirm="Reset" data-confirm-input="RESET">
```
Comment (Zaxounette) [inline BTCPayServer/Views/Manage/TwoFactorAuthentication.cshtml:62]:
```suggestion
                <a asp-action="GenerateRecoveryCodes" class="list-group-item d-flex justify-content-between align-items-center list-group-item-action py-3" data-bs-toggle="modal" data-bs-target="#ConfirmModal" data-title="Delete recovery codes" data-description="This action will delete the recovery codes from your BTCPay Server account. Your existing recovery codes will no longer be valid and you will have to set up new ones! Are you sure?" data-confirm="Delete" data-confirm-input="DELETE">
```

File: BTCPayServer/Views/Manage/TwoFactorAuthentication.cshtml [hunk 2]
```diff
@@ -3,83 +3,123 @@
...
+                    <div>
+                        <h5>Reset recovery codes</h5>
+                        <p class="mb-0 me-3">Regenerate your 2FA recovery codes.</p>
+                    </div>
+                    <vc:icon symbol="caret-right" />
+                </a>
+                <a asp-action="ResetAuthenticator" class="list-group-item d-flex justify-content-between align-items-center list-group-item-action py-3" data-bs-toggle="modal" data-bs-target="#ConfirmModal" data-title="Reset authenticator app" data-description="This process disables 2FA until you verify your authenticator app and will also reset your 2FA recovery codes. If you do not complete your authenticator app configuration you may lose access to your account." data-confirm="Reset" data-confirm-input="RESET">
```
Comment (Zaxounette) [inline BTCPayServer/Views/Manage/TwoFactorAuthentication.cshtml:69]:
```suggestion
                <a asp-action="ResetAuthenticator" class="list-group-item d-flex justify-content-between align-items-center list-group-item-action py-3" data-bs-toggle="modal" data-bs-target="#ConfirmModal" data-title="Reset authenticator app" data-description="This process disables 2FA until you verify your authenticator app and will also delete your 2FA recovery codes. If you do not complete your authenticator app configuration you may lose access to your account." data-confirm="Reset" data-confirm-input="RESET">
```

**Review by Zaxounette — COMMENTED** [review]
Some suggestions.

Feel free to discard them if I misunderstood the actual action being done in the backend.

**Comment by dennisreimann** [comment]
@pavlenex regarding point 2 I think this kinda defeats the initial purpose of unifying these cases. imho it works good as it is and otherwise we'd introduce yet another click to reach the security devices page.

I've updated the registering screen, now it looks like this:

fido2

**Comment by NicolasDorier** [comment]
Seems good to me @pavlenex  @Zaxounette ?

**Review by pavlenex — APPROVED** [review]
tACK
